manifests/server/config.pp

```diff
@@ -256,12 +257,14 @@
           content => template('postgresql/systemd-override.erb'),
           require => File['systemd-conf-dir'],
       ;
-
-      # Remove old unit file to avoid conflicts
-      'old-systemd-override':
-          ensure => absent,
-          path   => "/etc/systemd/system/${service_name}.service",
-      ;
     }
+
+    # Remove old unit file to avoid conflicts
+    file {'old-systemd-override':
+      ensure => absent,
+      owner  => root,
+      group  => root,
+      path   => "/etc/systemd/system/${service_name}.service",
+    } -> Exec['restart-systemd'] -> Class['postgresql::server::service']
```

Comment:
This implies that it's not triggered since there is no notify, correct?

I'd also suggest to set the `before` on the exec resource and rewrite this:
```suggestion
      before => Exec['restart-systemd'],
    }
```

Reply:
Thanks alot @ekohl for the review. Adding `before` will also enforce the ordering of resources to execute. If we add `before` here , we will also make changes to `restart-systemd` to have `before`, so that deletion of file, restart-systemd, starting of service are in order.